src/components/inputs/date-input/date-input.js

```diff
@@ -25,267 +25,235 @@ import {
   parseInputToDate,
 } from '../../../utils/calendar';
 
-class DateInput extends React.Component {
-  static displayName = 'DateInput';
+const DateInput = props => {
+  const intl = useIntl();
+  const [calendarDate, setCalendarDate] = React.useState(
+    props.value || getToday()
+  );
+  const [suggestedItems, setSuggestedItems] = React.useState([]);
+  const [highlightedIndex, setHighlightedIndex] = React.useState(
+    props.value === '' ? null : getDateInMonth(props.value) - 1
+  );
+  const inputRef = React.useRef();
 
-  static isEmpty = value => value === '';
-
-  static propTypes = {
-    intl: PropTypes.shape({
-      locale: PropTypes.string.isRequired,
-      formatMessage: PropTypes.func.isRequired,
-    }).isRequired,
-    horizontalConstraint: PropTypes.oneOf(['m', 'l', 'xl', 'scale']),
-    value: PropTypes.string.isRequired,
-    onChange: PropTypes.func.isRequired,
-    onFocus: PropTypes.func,
-    onBlur: PropTypes.func,
-    id: PropTypes.string,
-    name: PropTypes.string,
-    placeholder: PropTypes.string,
-    isDisabled: PropTypes.bool,
-    hasError: PropTypes.bool,
-    hasWarning: PropTypes.bool,
-  };
-
-  inputRef = React.createRef();
-
-  state = {
-    calendarDate: this.props.value || getToday(),
-    suggestedItems: [],
-    highlightedIndex:
-      this.props.value === '' ? null : getDateInMonth(this.props.value) - 1,
-  };
-
-  showPrevMonth = () => {
-    this.setState(prevState => ({
-      calendarDate: changeMonth(prevState.calendarDate, -1),
-      // select first day in next month
-      highlightedIndex: prevState.suggestedItems.length,
-    }));
-  };
-
-  jumpMonth = amount => {
-    this.setState(prevState => {
-      const nextDate = changeMonth(prevState.calendarDate, amount);
-      return { calendarDate: nextDate, highlightedIndex: 0 };
-    });
-  };
-
-  showToday = () => {
-    const today = getToday();
-    this.setState(
-      prevState => ({
-        calendarDate: today,
-        highlightedIndex:
-          prevState.suggestedItems.length + getDateInMonth(today) - 1,
+  const { onChange } = props;
```

Comment:
Same here

Reply:
Yeah, this is how I would go about it too. Thanks.